### ISSUE TITLE
Switch to more performant default key-value storage (dim rows, in particular)

### ISSUE THREAD

**Issue by cdeszaq** [yahoo/fili#293]
Currently the dimension row values are serialized to JSON which does not perform as well as some other options, and there's no human in the loop so a human-readable format like JSON is not needed. A better serialization/deserialization mechanism like kryo could improve the performance. 

Replace the JSON serialization and deserialization of dimension rows with a better performing serializer/deserializer.

**Comment by kevinhinterlong** [comment]
It may be easier to use [Smile](https://github.com/FasterXML/jackson-dataformats-binary/tree/master/smile)?

**Comment by cdeszaq** [comment]
Yeah, I'm not too opinionated on what serialization is used. All I know is that JSON isn't needed, and it's slower than other options. Ideally, I'd like to see some sort of benchmarking done to determine the route to use, and (possibly) as part of this, making the serializer / deserializer more easily swappable so that it could be changed out easily if someone's workload was different than the benchmarks tested.

Dimensions for the benchmarks:
- Cardinality (big dimensions vs small dimensions, where cardinalities range from 100,000,000 to 10)
- Size of field data (paragraphs vs. words, for example)
- Number of fields (1 field to 200 fields)
- Contents of fields (strings vs. numbers vs. dates, all as strings)
